trezorlib/tests/device_tests/test_msg_cardano_sign_transaction.py

```diff
@@ -153,3 +235,30 @@ def input_flow():
         )
         assert response.tx_hash.hex() == tx_hash
         assert response.tx_body.hex() == tx_body
+
+
+@pytest.mark.cardano
+@pytest.mark.skip_t1  # T1 support is not planned
+@setup_client()
+@pytest.mark.parametrize(
+    "protocol_magic,inputs,outputs,transactions,expected_error_message", INVALID_VECTORS
+)
+def test_cardano_sign_tx_validation(
+    client, protocol_magic, inputs, outputs, transactions, expected_error_message
+):
+    inputs = [cardano.create_input(i) for i in inputs]
+    outputs = [cardano.create_output(o) for o in outputs]
+
+    expected_responses = [
+        messages.CardanoTxRequest(tx_index=i) for i in range(len(transactions))
+    ]
+    expected_responses += [messages.Failure()]
+
+    with client:
+
+        client.set_expected_responses(expected_responses)
+
+        try:
+            cardano.sign_tx(client, inputs, outputs, transactions, protocol_magic)
+        except TrezorFailure as e:
```

Comment:
The test will pass if `sign_tx` does not throw. I believe that is not what we want ...

Reply:
fixed - added `assert False` outside the try-catch block. But perhaps you have a cleaner approach in mind

Reply:
Hmm, why not simply put assert inside the try block? That way it would be more obvious + it wouldn't break if we ever want to introduce a loop inside this code